src/tpm2-tss-engine-rand.c

```diff
@@ -67,12 +107,12 @@ rand_bytes(unsigned char *buf, int num)
         memcpy(buf, &b->buffer, b->size);
         num -= b->size;
         buf += b->size;
-        free(b);
```

Comment:
This change is incorrect. Esys_GetRandom allocates a new buffer on every invocation.

Reply:
the code in tpm2-tss, master, it's only new buffer while it's null. 

    Esys_GetRandom_Finish

/* Allocate memory for response parameters */
    if (randomBytes != NULL) {
        *randomBytes = calloc(sizeof(TPM2B_DIGEST), 1);
        if (*randomBytes == NULL) {
            return_error(TSS2_ESYS_RC_MEMORY, "Out of memory");
        }
    }